drivers/adc/adc_nrfx_saadc.c

```diff
@@ -719,6 +724,27 @@ static DEVICE_API(adc, adc_nrfx_driver_api) = {
 #endif
 };
 
+#ifdef NRF_SAADC_AIN8
```

Comment:
I think we should be consistent and use `#if defined(CONFIG_NRF_PLATFORM_HALTIUM)` also here.